app/controllers/subjects_controller.rb

```diff
@@ -4,16 +4,18 @@ class SubjectsController < ApplicationController
   # do matter, and changing them will affect the breadcrumb order in the UI.
   add_breadcrumb :search_filter_breadcrumb_name, :new_search_filter_path
   add_breadcrumb :application_search_breadcrumb_name, :search_breadcrumb_path
-  before_action :load_and_authorize_application, :load_link
+  before_action :load_and_authorize_application
 
-  def show; end
+  def show
+    @form_model = @link_status.maat_linked? ? load_unlink_attempt : load_link_attempt
```

Comment:
This is to make things more explicit and clear to understand, instead of checking the code on the `before_action`.